src/t-i18n.ts

```diff
@@ -99,29 +99,29 @@ export class I18n {
 }
 
 function createT(context: I18n): TFunc {
-	let T: any = function translate(message: string, replacements?: (Replacements), id?: string): string {
+	const T = (message: string, replacements?: IcuReplacements, id?: string): string => {
 		if (!id) id = context.generateId(message);
 		return context.lookup(id, replacements, message);
 	}
-	T._i18nInstance = context;
-	T.setup = context.setup.bind(T._i18nInstance);
-	T.lookup = context.lookup.bind(T._i18nInstance);
-	T.date = context.format.bind(T._i18nInstance, "date");
-	T.number = context.format.bind(T._i18nInstance, "number");
-
-	T.$ = function translateReact(message: string, replacements: ReactReplacements = null, id?: string): React.ReactNode[] {
-		const translatedMessage = T.apply(this, arguments);
-		const reactElements = parseReact(translatedMessage, replacements);
-		return reactElements;
-
-	}
-
-	return T;
+	const properties = {
+		_i18nInstance: context,
+		setup: context.setup.bind(context),
+		lookup: context.lookup.bind(context),
+		date: context.format.bind(context, "date"),
+		number: context.format.bind(context, "number"),
+		$: <T>(message: string, replacements: AnyReplacements<T> = {}, id?: string): (T | string)[] => {
+			const [icu, xml] = splitReplacements(replacements);
+			const translatedMessage = T(message, icu, id);
+			return parseXml(translatedMessage, xml);
+		},
+	};
+
+	return assign(T, properties);
```

Comment:
It doesn't work with `Object.assign`?

Reply:
`Object.assign` is not available in ES5, and TS does not compile it to ES5 on its own.